File: iosApp/CocktailCraft/CocktailCraft/Theme/Theme.swift
import SwiftUI

// MARK: - App Colors
struct AppColors {
    // Light Theme Colors
    static let primaryLight = Color(red: 0.922, green: 0.416, blue: 0.263) // #EB6A43
    static let secondaryLight = Color(red: 1.0, green: 0.784, blue: 0.302) // #FFC84D
    static let backgroundLight = Color(red: 0.98, green: 0.98, blue: 0.98) // #FAFAFA
    static let surfaceLight = Color.white
    static let textPrimaryLight = Color.black
    static let textSecondaryLight = Color(red: 0.557, green: 0.557, blue: 0.576) // #8E8E93
    
    // Dark Theme Colors
    static let primaryDark = Color(red: 1.0, green: 0.541, blue: 0.396) // #FF8A65
    static let secondaryDark = Color(red: 1.0, green: 0.82, blue: 0.502) // #FFD180
    static let backgroundDark = Color(red: 0.071, green: 0.071, blue: 0.071) // #121212
    static let surfaceDark = Color(red: 0.118, green: 0.118, blue: 0.118) // #1E1E1E
    static let textPrimaryDark = Color.white
    static let textSecondaryDark = Color(red: 0.69, green: 0.69, blue: 0.69) // #B0B0B0
    
    // Common Colors (same in both themes)
    static let success = Color(red: 0.204, green: 0.78, blue: 0.349) // #34C759
    static let error = Color(red: 1.0, green: 0.231, blue: 0.188) // #FF3B30
    static let warning = Color(red: 1.0, green: 0.584, blue: 0) // #FF9500
    static let gray = Color(red: 0.557, green: 0.557, blue: 0.576) // #8E8E93
    static let darkGray = Color(red: 0.388, green: 0.388, blue: 0.4) // #636366
    static let lightGray = Color(red: 0.898, green: 0.898, blue: 0.918) // #E5E5EA
    static let chipBackground = Color(red: 0.612, green: 0.361, blue: 0.22) // #9C5C38
}

// MARK: - Color Scheme Environment Key
private struct ColorSchemeKey: EnvironmentKey {
    static let defaultValue = ColorScheme.light
}

extension EnvironmentValues {
    var appColorScheme: ColorScheme {
        get { self[ColorSchemeKey.self] }
        set { self[ColorSchemeKey.self] = newValue }
    }
}

// MARK: - Theme-aware Color Extensions
extension Color {
    // Dynamic colors that change based on theme
    static func primary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.primaryDark : AppColors.primaryLight
    }
    
    static func secondary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.secondaryDark : AppColors.secondaryLight
    }
    
    static func background(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.backgroundDark : AppColors.backgroundLight
    }
    
    static func surface(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.surfaceDark : AppColors.surfaceLight
    }
    
    static func textPrimary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.textPrimaryDark : AppColors.textPrimaryLight
    }
    
    static func textSecondary(for colorScheme: ColorScheme) -> Color {
        colorScheme == .dark ? AppColors.textSecondaryDark : AppColors.textSecondaryLight
    }
}

// MARK: - View Modifiers for Theme Support
struct ThemedBackground: ViewModifier {
    @Environment(\.colorScheme) var colorScheme
    
    func body(content: Content) -> some View {
        content
            .background(Color.background(for: colorScheme))
    }
}

struct ThemedCard: ViewModifier {
    @Environment(\.colorScheme) var colorScheme
    let cornerRadius: CGFloat
    
    func body(content: Content) -> some View {
        content
            .background(Color.surface(for: colorScheme))
            .cornerRadius(cornerRadius)
            .shadow(color: Color.black.opacity(colorScheme == .dark ? 0.3 : 0.1), radius: 2, x: 0, y: 1)
    }
}

// MARK: - Convenience View Extensions
extension View {
    func themedBackground() -> some View {
        modifier(ThemedBackground())
    }
    
    func themedCard(cornerRadius: CGFloat = 12) -> some View {
        modifier(ThemedCard(cornerRadius: cornerRadius))
    }
    
    func primaryButton(colorScheme: ColorScheme) -> some View {
        self
            .foregroundColor(.white)
            .background(Color.primary(for: colorScheme))
            .cornerRadius(8)
    }
    
    func secondaryButton(colorScheme: ColorScheme) -> some View {
        self
            .foregroundColor(Color.primary(for: colorScheme))
            .overlay(
                RoundedRectangle(cornerRadius: 8)
                    .stroke(Color.primary(for: colorScheme), lineWidth: 1)
            )
    }
}

// MARK: - Theme Manager
class ThemeManager: ObservableObject {
    static let shared = ThemeManager()
    
    @Published var currentTheme: ColorScheme = .light
    
    private init() {}
    
    func updateTheme(isDarkMode: Bool) {
        currentTheme = isDarkMode ? .dark : .light
    }
}

File: androidApp/src/main/java/com/cocktailcraft/android/ui/theme/Theme.kt
package com.cocktailcraft.android.ui.theme

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.compose.ui.graphics.Color
import com.cocktailcraft.designsystem.AccentColorTokens
import com.cocktailcraft.designsystem.ColorTokens

// CocktailCraft color palette, mapped 1:1 from the cross-platform design
// tokens in shared/designsystem/DesignTokens.kt — the single source of truth
// for BOTH Android and iOS. res/values/colors.xml (pre-Compose window +
// widget assets) still mirrors these values by hand; keep it in sync when
// changing anything in the shared tokens.
//
// AN-5 status: the dead CocktailBarTheme (and its deprecated
// window.statusBarColor handling) was deleted — AnimatedCocktailBarTheme in
// AnimatedTheme.kt is the app's one theme. The remaining follow-up is the
// incremental migration of AppColors.* reads (~320 call sites) onto
// MaterialTheme.colorScheme, after which the mutable isDarkTheme global goes
// away and dynamic color can be considered.
object AppColors {
    // Light Theme Colors
    val PrimaryLight = Color(ColorTokens.primaryLight) // Coral/Orange for main elements
    val SecondaryLight = Color(ColorTokens.secondaryLight) // Yellow/Gold for wave and accents
    val BackgroundLight = Color(ColorTokens.backgroundLight) // Light background
    val SurfaceLight = Color(ColorTokens.surfaceLight) // White surface
    val TextPrimaryLight = Color(ColorTokens.textPrimaryLight) // Black for primary text
    val TextSecondaryLight = Color(ColorTokens.textSecondaryLight) // Gray for secondary text

    // Dark Theme Colors
    val PrimaryDark = Color(ColorTokens.primaryDark) // Lighter coral for dark theme
    val SecondaryDark = Color(ColorTokens.secondaryDark) // Lighter gold for dark theme
    val BackgroundDark = Color(ColorTokens.backgroundDark) // Dark background
    val SurfaceDark = Color(ColorTokens.surfaceDark) // Dark surface
    val TextPrimaryDark = Color(ColorTokens.textPrimaryDark) // White for primary text in dark mode
    val TextSecondaryDark = Color(ColorTokens.textSecondaryDark) // Light gray secondary text in dark mode

    // Common Colors (same in both themes)
    val Success = Color(ColorTokens.success) // Green for success states
    val Error = Color(ColorTokens.error) // Red for errors
    val Warning = Color(ColorTokens.warning) // Orange for warnings and stars
    val Gray = Color(ColorTokens.gray) // Gray for secondary text
    val DarkGray = Color(ColorTokens.darkGray) // Dark gray for stronger text
    val LightGray = Color(ColorTokens.lightGray) // Light gray for backgrounds
    val ChipBackground = Color(ColorTokens.chipBackground) // Brown for category chips like "Vodka"

    // Dynamic colors based on theme. Snapshot-backed so any composable that
    // reads AppColors.* subscribes and recomposes when the theme flips —
    // a plain var here silently served stale colors after theme changes.
    // All three are written only by AnimatedCocktailBarTheme from the shared
    // theme state.
    var isDarkTheme by mutableStateOf(false)
    var accentName by mutableStateOf(AccentColorTokens.DEFAULT)
    var isHighContrast by mutableStateOf(false)

    // Primary follows the user-selected accent ("coral" = brand default)
    val Primary get() = Color(
        if (isDarkTheme) AccentColorTokens.dark(accentName)
        else AccentColorTokens.light(accentName)
    )
    val Secondary get() = if (isDarkTheme) SecondaryDark else SecondaryLight
    val Background get() = if (isDarkTheme) BackgroundDark else BackgroundLight
    val Surface get() = if (isDarkTheme) SurfaceDark else SurfaceLight
    val TextPrimary get() = if (isDarkTheme) TextPrimaryDark else TextPrimaryLight

    // High contrast pushes secondary text further from the background
    val TextSecondary get() = when {
        isHighContrast && isDarkTheme -> LightGray
        isHighContrast -> DarkGray
        isDarkTheme -> TextSecondaryDark
        else -> TextSecondaryLight
    }
}
